lib/internal/process/promises.js

```diff
@@ -215,137 +338,179 @@ function emitUnhandledRejectionWarning(uid, reason) {
+      promise,
+    );
+  }
+  const handled = emitUnhandledRejection(promise, promiseInfo);
+  if (!handled) emitUnhandledRejectionWarning(promiseInfo);
+  return true;
+}
+
+/**
+ * --unhandled-rejections=none:
+ * Emit 'unhandledRejection', but do not emit any warning.
+ * @type {UnhandledRejectionsModeHandler}
+ */
+function ignoreUnhandledRejectionsMode(promise, promiseInfo) {
+  emitUnhandledRejection(promise, promiseInfo);
+  return true;
+}
+
+/**
+ * --unhandled-rejections=warn:
+ * Emit 'unhandledRejection', then emit 'UnhandledPromiseRejectionWarning'.
+ * @type {UnhandledRejectionsModeHandler}
+ */
+function alwaysWarnUnhandledRejectionsMode(promise, promiseInfo) {
+  emitUnhandledRejection(promise, promiseInfo);
+  emitUnhandledRejectionWarning(promiseInfo);
+  return true;
+}
+
+/**
+ * --unhandled-rejections=throw:
+ * Emit 'unhandledRejection', if it's unhandled, emit
+ * 'uncaughtException'. If it's not handled, print the error to stderr
+ * and exit the process.
+ * @type {UnhandledRejectionsModeHandler}
+ */
+function throwUnhandledRejectionsMode(promise, promiseInfo) {
+  const reason = promiseInfo.reason;
+  const handled = emitUnhandledRejection(promise, promiseInfo);
+  if (!handled) {
+    const err = isErrorLike(reason) ?
+      reason :
+      new UnhandledPromiseRejection(reason);
+    // This destroys the async stack, don't clear it after
+    triggerUncaughtException(err, true /* fromPromise */);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * --unhandled-rejections=warn-with-error-code:
+ * Emit 'unhandledRejection', if it's unhandled, emit
+ * 'UnhandledPromiseRejectionWarning', then set process exit code to 1.
+ * @type {UnhandledRejectionsModeHandler}
+ */
+function warnWithErrorCodeUnhandledRejectionsMode(promise, promiseInfo) {
+  const handled = emitUnhandledRejection(promise, promiseInfo);
+  if (!handled) {
+    emitUnhandledRejectionWarning(promiseInfo);
+    process.exitCode = kGenericUserError;
+  }
+  return true;
+}
+
+/**
+ * @returns {UnhandledRejectionsModeHandler}
+ */
+function getUnhandledRejectionsMode() {
+  const { getOptionValue } = require('internal/options');
+  switch (getOptionValue('--unhandled-rejections')) {
+    case 'none':
+      return ignoreUnhandledRejectionsMode;
+    case 'warn':
+      return alwaysWarnUnhandledRejectionsMode;
+    case 'strict':
+      return strictUnhandledRejectionsMode;
+    case 'throw':
+      return throwUnhandledRejectionsMode;
+    case 'warn-with-error-code':
+      return warnWithErrorCodeUnhandledRejectionsMode;
+    default:
+      return throwUnhandledRejectionsMode;
+  }
+}
+
 // If this method returns true, we've executed user code or triggered
 // a warning to be emitted which requires the microtask and next tick
 // queues to be drained again.
 function processPromiseRejections() {
-  let maybeScheduledTicksOrMicrotasks = asyncHandledRejections.length > 0;
+  let maybeScheduledTicksOrMicrotasks = asyncHandledRejections.length !== 0;
```

Comment:
Nit (no need to address here): For me it's helpful to keep all lines as they are besides the ones that need to be changed. That way I am able to review things faster and it is easier to understand what really changed. I guess it's a personal preference as it has no impact on the performance, or change the functionality?

Reply:
Actually this is a personal quirk which I picked up like 20 years ago, when reading about C. Not equal 0 is faster than greater than 0, as it has only to check for one specific value instead of checking if the number is actually greater or not. .length can not be smaller than 0. 

As I touched nearly every line, I guess I also touched that line with having above considerations in mind. Can I prove it via benchmark? I guess not.

Reply:
reverted that line